File: eau08.exs
defmodule Exercice do
  def validate_args([arg]) when bit_size(arg) > 0, do: {:ok, arg}
  def validate_args(_args), do: :error

  def number_only?(<<>>), do: true
  def number_only?(<<char, _tail::binary>>) when char not in ?0..?9, do: false
  def number_only?(<<_number_char, tail::binary>>), do: number_only?(tail)
end

System.argv()
|> Exercice.validate_args()
|> case do
  {:ok, arg} ->
    arg
    |> Exercice.number_only?()
    |> IO.puts()

  :error ->
    IO.puts("usage: elixir eau08.exs")
end

File: eau08.go
package main

// Chiffre only
import (
	"fmt"
	"os"
)

func numberOnly(input string) bool {
	for _, char := range input {
		if char < '0' || char > '9' {
			return false
		}

	}
	return true
}

func main() {
	args := os.Args[1:]
	if len(args) != 1 {
		fmt.Println("usage: go run eau08.go")
		return
	}

	input := args[0]
	fmt.Println(numberOnly(input))
}
